spring-boot/src/main/java/com/sentrydemos/springboot/Application.java

```diff
@@ -44,24 +44,15 @@ class CustomTracesSamplerCallback implements TracesSamplerCallback {
 		public Double sample(SamplingContext context) {
 
 			CustomSamplingContext customSamplingContext = context.getCustomSamplingContext();
-			if (customSamplingContext != null) {
```

Comment:
are you sure we should remove this null check here? What is is null one day? Then I think the following line would error:
```
HttpServletRequest request = (HttpServletRequest) customSamplingContext.get("request");
```

Reply:
if customSamplingContext is null/undefined then it will say "can't read .get of undefined".

java likes to do a lot of null checks, and there's a good chance whatever boilerplate code I got that from a while ago, it probably gave me that null check. I tend to not insert them myself.

Reply:
sounds good.